### PR TITLE
Add AWS DynamoDB Spy

### LINKED ISSUE #826: Add Spy for DynamoDB
I would like to add support for [AWS DynamoDB](https://github.com/aws/aws-sdk-ruby/tree/master/gems/aws-sdk-dynamodb) by adding a spy.

I already got a working spy going. I'm just working on adding tests now.

### PR DESCRIPTION
**Pull request by pbstriker38**
## What does this pull request do?

Add support for DynamoDB

## Checklist

- [x] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/). 
- [ ] My code follows the style guidelines of this project (See `.rubocop.yml`)
- [x] I have rebased my changes on top of the latest master branch

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing [**unit** tests](https://github.com/elastic/apm-agent-ruby/blob/master/CONTRIBUTING.md#testing) pass locally with my changes

- [ ] I have made corresponding changes to the documentation
- [x] I have updated [CHANGELOG.asciidoc](CHANGELOG.asciidoc)
- [x] I have updated [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)
- [ ] Added an API method or config option? Document in which version this will be introduced

## Related issues
- Closes #826

### CODE REVIEW

**Comment by mikker** [comment]
jenkins, please run the tests

**Comment by mikker** [comment]
This looks just wonderful! Have you tried this in production? Verified that it works as intended?

It feels a bit extensive to wrap each api method on each own but as far as I understand it, they all use a method from aws-sdk-core internally, which we don't want to wrap. Is that correct?

**Comment by pbstriker38** [comment]
Yes. I have tried this out on a Rack + Grape app.

image
image
image
image

**Comment by pbstriker38** [comment]
Yes. They all use the `Seahorse` client from aws-sdk-core. I was initially only going to wrap `get_item` and `put_item` since that is what we are using, but to make this useful for everyone, I thought it would be best to wrap all DynamoDB operations.

It looks like the AWS SDK is trying to reach out to `http://169.254.169.254/latest/api/token` in CI. Let me fix that.

**Comment by mikker** [comment]
Thanks! I'm not familiar with DynamoDB but I assume there are statements or queries? Does it make sense to include those like we do for Elasticsearch or SQL-based adapters?

**Comment by pbstriker38** [comment]
Yes. Each of the operations takes params. If you want me to include those as context, that is doable. For our use case though, we would actually want to disable that feature, or filter most of it.

**Comment by mikker** [comment]
Let's start without it then 👍 we can always add it later on if someone wants it.

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -35,6 +35,11 @@ endif::[]
 [[release-notes-3.x]]
 === Ruby Agent version 3.x
 
+[float]
```
Comment (mikker) [inline CHANGELOG.asciidoc:38]:
```suggestion
[float]
==== Unreleased

[float]
```

We usually include this header.

File: docs/supported-technologies.asciidoc [hunk 2]
```diff
@@ -151,4 +152,4 @@ To instrument a server, add the `ElasticAPM::GRPC::ServerInterceptor`.
 [source,ruby]
 ----
 GRPC::RpcServer.new(interceptors: [ElasticAPM::GRPC::ServerInterceptor.new])
-----
\ No newline at end of file
+----
```
Comment (mikker) [inline docs/supported-technologies.asciidoc:155]: 👏

**Comment by mikker** [comment]
jenkins, run the tests

**Comment by mikker** [comment]
Test failures are unrelated, fix in https://github.com/elastic/apm-agent-ruby/pull/828

**Comment by mikker** [comment]
Thank you so much for your contribution 💙💚💛💜❤️

**Pull request merged**